src/nidcpower/system_tests/test_system_nidcpower.py

```diff
@@ -458,19 +487,211 @@ def test_init_raises_value_error_for_multi_instrument_resource_name_and_channels
+def test_init_raises_driver_errors_for_invalid_arguments(resource_name, channels):
     """Tests for driver errors that should occur for invalid initialization arguments."""
     options = {'Simulate': True, 'DriverSetup': {'Model': '4162', 'BoardType': 'PXIe'}}
     with pytest.raises(nidcpower.errors.DriverError) as e:
-        with nidcpower.Session(resource_name, channels, options=options, independent_channels=independent_channels) as session:
-            # multi-instrument resource names are valid for simulated initialize with channels
-            # sessions, so we make a driver call on channels and ensure that errors
-            session.channels[channels].output_function = nidcpower.OutputFunction.DC_VOLTAGE
-    assert e.value.code == expected_error_code
+        with nidcpower.Session(resource_name, channels, options=options):
+            pass
+    assert e.value.code == -1074097793
+    assert e.value.description.find('The specified device cannot be found.') != -1
+
+
+@pytest.mark.parametrize(
+    'resource_name,channels',
+    [
+        ('Dev1/0', None),
+        ('Dev1/0', 'Dev1/0'),
+        ('Dev1/0,Dev2/0', 'Dev1/0'),
+        ('Dev1,Dev2', '')
+    ]
+)
+def test_init_raises_driver_errors_for_invalid_arguments_legacy_session(resource_name, channels):
+    """Tests invalid initializer arguments for legacy initialize with channels sessions.
+
+    Multi-instrument resource names are valid for simulated initialize with channels sessions. So,
+    we attempt to initialize a true session and assert an unsupported device exception is raised.
+    """
+
+    with pytest.raises(nidcpower.errors.DriverError) as e:
+        with nidcpower.Session(resource_name, channels, independent_channels=False):
+            pass
+    assert e.value.code == -1074118656
+    assert e.value.description.find('The device is not supported with this driver or version.') != -1
+
+
+@pytest.mark.include_legacy_session
+def test_repeated_capabilities_on_method_when_all_channels_are_specified(session):
+    """Sessions should not error when specifying all channels by number."""
+    assert session.channels['0'].output_enabled is True
+    session.channels['0'].output_enabled = False
+    session.channels['0-11'].reset()
+    assert session.channels['0'].output_enabled is True
+
+
+@pytest.mark.legacy_session_only
+def test_error_channel_name_not_allowed_in_legacy_session(session):
+    with pytest.raises(nidcpower.Error) as e:
+        session.channels['0'].reset()
+    assert e.value.code == -1074118494  # NIDCPOWER_ERROR_CHANNEL_NAME_NOT_ALLOWED_IN_OBSOLETE_SESSION
+    assert e.value.description.find('The channel name string must represent all channels in the session because the session was not initialized with independent channels. To specify a subset of channels for this function, first initialize the session with independent channels.') != -1
+
+
+@pytest.mark.legacy_session_only
+def test_error_channel_name_not_allowed(session):
```

Comment:
I think this test works with Independent Channels session too. Any particular reason for using `legacy_session_only`?

Reply:
For legacy sessions, `instrument_model` isn't expecting a channel string to be supplied.
For independent channels sessions, it expects the instrument name to be supplied.

So for an independent channels session, it throws a different error:
`"The specified device name cannot be found."`

Do you think we should also test for the independent channels error?

Reply:
Ah, you're right. On the Independent Channels session, Python layer doesn't really differentiate between the various repeated capability containers and passes it down to the driver Runtime. So, we don't need to test it with  Independent Channels session.